tests/tracer/test_sampling_decision.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from ddtrace.context import Context
+from ddtrace.internal.sampling import SamplingMechanism
+from ddtrace.internal.sampling import _hash_service
+from ddtrace.internal.sampling import update_sampling_decision
+from tests.utils import override_global_config
+
+
+@pytest.fixture()
+def context():
+    yield Context()
+
+
+@pytest.mark.parametrize(
+    "service,expected",
+    [
+        ("service", "9df6b026a8"),
+        (u"\x80service", "069dd5df76"),
+        (b"\x80service", None),
+    ],
+)
+def test_hash_service(service, expected):
+    assert expected == _hash_service(service)
+
+
+@pytest.mark.parametrize(
+    "propagate_service,service,sampling_mechanism,sampled,expected",
+    [
+        (True, None, SamplingMechanism.DEFAULT, True, "-0"),
+        (True, None, SamplingMechanism.AGENT_RATE, True, "-1"),
+        (True, None, SamplingMechanism.TRACE_SAMPLING_RULE, True, "-3"),
+        (True, None, SamplingMechanism.MANUAL, True, "-4"),
+        (True, "service", SamplingMechanism.DEFAULT, True, "9df6b026a8-0"),
+        (True, "service", SamplingMechanism.AGENT_RATE, True, "9df6b026a8-1"),
+        (True, "service", SamplingMechanism.TRACE_SAMPLING_RULE, True, "9df6b026a8-3"),
+        (True, "service", SamplingMechanism.MANUAL, True, "9df6b026a8-4"),
+        (False, "service", SamplingMechanism.DEFAULT, True, "-0"),
+        (True, "service", SamplingMechanism.DEFAULT, False, None),
```

Comment:
This final one is `None` for the `expected` value, because since it isn't sampled, we remove the tag.